packages/plugin-npm/tests/npmPublishUtils.ts

```diff
@@ -0,0 +1,48 @@
+import {httpUtils}         from '@yarnpkg/core';
```

Comment:
Few things that would really help me -

1. About tests :
  -  I have two ideas how to implement:
  - e2e test with `makeTemporaryEnv`, spyOn `npmHttpUtils.put` and expect to be called with body that contain readme.
  - unit test and builders, build required params for `makePublishBody` aka 
  `workspace: Workspace, buffer: Buffer, {access, tag, registry, gitHead}: PublishAdditionalParams`
  such as inside builders.ts
  `export function aWorkspace(): Workspace{}` 
  and just call this function.
  
I think the first option is better but I can't seem to succeed implement a spyOn.
any suggestions ?

2. I believe `readmeFilename` is deprecated but -
see this -
https://github.com/npm/npm/issues/3573#issuecomment-19550501
and this -
https://github.com/npm/cli/issues/4252

Do you think we need to add this ?

Reply:
Also should add test for workspace(monorepo) as the readme file is read with cwd,
I don't know if cwd is relative when using `yarn workspaces foreach npm publish`